app/models/miq_queue.rb

```diff
@@ -696,19 +696,18 @@ def self.optional_values(options, keys = [:zone])
     return unless ENV["MESSAGING_HOSTNAME"] && ENV["MESSAGING_PORT"] && ENV["MESSAGING_USERNAME"] && ENV["MESSAGING_PASSWORD"]
 
     options = {
-      :host     => ENV["MESSAGING_HOSTNAME"],
-      :port     => ENV["MESSAGING_PORT"].to_i,
-      :username => ENV["MESSAGING_USERNAME"],
-      :password => ENV["MESSAGING_PASSWORD"],
-      :protocol => ENV.fetch("MESSAGING_PROTOCOL", "Kafka"),
-      :encoding => ENV.fetch("MESSAGING_ENCODING", "json")
+      :host           => ENV["MESSAGING_HOSTNAME"],
+      :port           => ENV["MESSAGING_PORT"].to_i,
+      :username       => ENV["MESSAGING_USERNAME"],
+      :password       => ENV["MESSAGING_PASSWORD"],
+      :protocol       => ENV.fetch("MESSAGING_PROTOCOL", "Kafka"),
+      :encoding       => ENV.fetch("MESSAGING_ENCODING", "json"),
+      :sasl_mechanism => ENV.fetch("MESSAGING_SASL_MECHANISM", "PLAIN")
     }
 
-    if ENV["MESSAGING_KEYSTORE_PASSWORD"].present?
+    if ENV["MESSAGING_SSL_CA"].present?
       options[:ssl] = true
```

Comment:
This can be a future PR but I wonder if we should split up "SSL" from "CA_FILE".
I could see for development especially we might want to use ssl but not verify the peer cert

Reply:
I'll make note for a future PR